skypy/halos/tests/test_sham.py

```diff
@@ -0,0 +1,1209 @@
+        z_range = [0.01, 0.1]
+        skyarea = 600
+        min_mass = 10**(7.5)
+        max_mass = 10**(14)
+        file_name = 'unit_test.yaml'
+
+        # Function
+        cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range, skyarea,
+                         min_mass, max_mass, file_name)
+        assert str(excinfo.value) == 'M* and phi* must be positive and non-zero numbers'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        m_star, phi_star, alpha = -10**(10.58), 0, -0.33
+        cosmology = FlatLambdaCDM(H0=70, Om0=0.3, name='FlatLambdaCDM')
+        z_range = [0.01, 0.1]
+        skyarea = 600
+        min_mass = 10**(7.5)
+        max_mass = 10**(14)
+        file_name = 'unit_test.yaml'
+
+        # Function
+        cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range, skyarea,
+                         min_mass, max_mass, file_name)
+        assert str(excinfo.value) == 'M* and phi* must be positive and non-zero numbers'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        m_star, phi_star, alpha = 10**(10.58), 10**(-2.77), -0.33
+        cosmology = FlatLambdaCDM(H0=70, Om0=0.3, name='FlatLambdaCDM')
+        z_range = [0.01, 0.1]
+        skyarea = -6
+        min_mass = 10**(7.5)
+        max_mass = 10**(14)
+        file_name = 'unit_test.yaml'
+
+        # Function
+        cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range, skyarea,
+                         min_mass, max_mass, file_name)
+        assert str(excinfo.value) == 'The skyarea must be a positive non-zero number'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        m_star, phi_star, alpha = 10**(10.58), 10**(-2.77), -0.33
+        cosmology = FlatLambdaCDM(H0=70, Om0=0.3, name='FlatLambdaCDM')
+        z_range = [0.01, 0.1]
+        skyarea = 600
+        min_mass = 10**(14)
+        max_mass = 10**(8)
+        file_name = 'unit_test.yaml'
+
+        # Function
+        cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range, skyarea,
+                         min_mass, max_mass, file_name)
+        assert str(excinfo.value) == 'The minimum mass should be less than the maximum mass'
+
+    with pytest.raises(Exception) as excinfo:
+        # Parameters
+        m_star, phi_star, alpha = 10**(10.58), 10**(-2.77), -0.33
+        cosmology = FlatLambdaCDM(H0=70, Om0=0.3)
+        z_range = [0.01, 0.1]
+        skyarea = 600
+        min_mass = 10**(7.5)
+        max_mass = 10**(14)
+        file_name = 'unit_test.yaml'
+
+        # Function
+        cat = galaxy_cat(m_star, phi_star, alpha, cosmology, z_range, skyarea,
+                         min_mass, max_mass, file_name)
+        assert str(excinfo.value) == 'Cosmology object must have an astropy cosmology name'
+
+    file.close()
+    os.remove('unit_test.yaml')
+
+
+@pytest.mark.skipif(not HAS_COLOSSUS, reason='test requires colossus')
+@pytest.mark.flaky
+def test_assignment():

```

Comment:
Can you please add comments (just short ones) describing what each test is for right underneath the `def`?